vars/mavenTemplate.groovy

```diff
@@ -9,14 +9,14 @@ def call(Map parameters = [:], body) {
     def defaultLabel = buildId('maven')
     def label = parameters.get('label', defaultLabel)
 
-    def mavenImage = parameters.get('mavenImage', 'fabric8/maven-builder:v9ff62e0')
-    def jnlpImage = (flow.isOpenShift()) ? 'fabric8/jenkins-slave-base-centos7:v2132422' : 'jenkinsci/jnlp-slave:2.62'
+    def mavenImage = parameters.get('mavenImage', 'openshift/jenkins-slave-maven-centos7:v4.0')
+    def jnlpImage = (flow.isOpenShift()) ? 'openshift/jenkins-slave-base-centos7:v4.0' : 'jenkinsci/jnlp-slave:2.62'
     def inheritFrom = parameters.get('inheritFrom', 'base')
 
 
     def cloud = flow.getCloudConfig()
 
-    def javaOptions = parameters.get('javaOptions', '-Duser.home=/root/ -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -Dsun.zip.disableMemoryMapping=true -XX:+UseParallelGC -XX:MinHeapFreeRatio=5 -XX:MaxHeapFreeRatio=10 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Xms10m -Xmx192m')
+    def javaOptions = parameters.get('javaOptions', '-Duser.home=/home/jenkins -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -Dsun.zip.disableMemoryMapping=true -XX:+UseParallelGC -XX:MinHeapFreeRatio=5 -XX:MaxHeapFreeRatio=10 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Xms10m -Xmx192m')
```

Comment:
does env.JENKINS_HOME work? instead of hardcoded `/home/jenkins`

Reply:
@sthaha I will test that out and see if we can switch to that but previously this was just set to /root, but I agree it would be better to use the env.JENKINS_HOME

Reply:
@sthaha actually this should just be ~/ not env.JENKINS_HOME since it is not the Jenkins home directory where this needs to be stored.  That is probably why it was /root first.

Reply:
@sthaha /home/jenkins is the right approach and should not be ~/

Reply:
@sthaha comment added explaining why we need to use /home/jenkins